src/main/scala/zio/keeper/Cluster.scala

```diff
@@ -0,0 +1,46 @@
+package zio.keeper
+
+import java.net.InetAddress
+
+import zio.{ IO, ZIO }
+import zio.stream.Stream
+import zio.keeper.Cluster._
+
+trait Cluster {
+  final def join: ZIO[Credentials with Discovery with Transport with Metadata, Error, Handle] = ???
+}
+
+object Cluster {
+  // TODO: properly represent (see current 'develop' branch)
+  type Path[_, _]
+  type Type[_]
+
+  object Type {
+    def apply[A](implicit instance: Type[A]): Type[A] = instance
+  }
+
+  trait Credentials {
+    // TODO: ways to obtain auth data
+  }
+
+  trait Discovery {
+    def discover: IO[Error, Set[InetAddress]]
```

Comment:
The problem with this interface is that it does not permit notification on new members, etc. (in which case, `Managed[Error, Stream[Error, MembershipEvent]]` could work). Also possibly we should introduce the notion of "channel"—perhaps that's a "key" provided to "discover", i.e. give me all the nodes that identify with the specified key.

Reply:
this is interface that only provide list of seeds of cluster. If we want to use this in any production scenario we need to provide some way to discover them. typically it's not as easy as provide list of the ips. for instance when you are using kubernetes that will be call to kubernetes's dns or service for list of nodes.

Reply:
OK, that makes sense. 👍

Reply:
As @pshemass explained, our rough idea is to handle `MembershipEvent` event stream internally.